pip/pep425tags.py

```diff
@@ -124,7 +124,54 @@ def get_platform():
         split_ver = release.split('.')
         return 'macosx_{0}_{1}_{2}'.format(split_ver[0], split_ver[1], machine)
     # XXX remove distutils dependency
-    return distutils.util.get_platform().replace('.', '_').replace('-', '_')
+    result = distutils.util.get_platform().replace('.', '_').replace('-', '_')
+    if result == "linux_x86_64" and sys.maxsize == 2147483647:
+        result = "linux_i686"
+    return result
+
+
+def is_manylinux1_compatible():
+    # Only Linux, and only x86-64 / i686
+    if get_platform() not in ("linux_x86_64", "linux_i686"):
+        return False
+
+    # Check for presence of _manylinux module
+    try:
+        import _manylinux
+        return bool(_manylinux.manylinux1_compatible)
+    except (ImportError, AttributeError):
+        # Fall through to heuristic check below
+        pass
+
+    # Check glibc version. CentOS 5 uses glibc 2.5.
+    return have_compatible_glibc(2, 5)
+
+
+def have_compatible_glibc(major, minimum_minor):
+    import ctypes
+    process_namespace = ctypes.CDLL(None)
```

Comment:
@dstufft: Yeah, that's exactly right -- `ctypes.CDLL(None)` internally calls `dlopen(NULL)`, and as the `dlopen` manpage says, "If *filename* is NULL, then the returned handle is for the main program". So this is a neat way to dodge the whole problem of figuring out which libc our process is actually using -- the linker already knows that, so we let it do the work :-).